pkg/reconciler/revision/resources/pa.go

```diff
@@ -48,6 +48,8 @@ func MakePA(rev *v1alpha1.Revision) *av1alpha1.PodAutoscaler {
 				Kind:       "Deployment",
 				Name:       names.Deployment(rev),
 			},
+			// TODO(tanzeeb) Tests!
+			Reachable:    rev.HasRoute(),
```

Comment:
One thing I noticed with this change is that this causes us to change the status on scaled-to-zero Revisions, and they basically get stuck in `Updating`: https://github.com/knative/serving/blob/c6f7bc48faa2c882539d56719d5b23fdd653ea5c/pkg/reconciler/revision/reconcile_resources.go#L169-L171

Given that such an update for an otherwise inactive Revision cannot change the Revision's status back to Ready, I'm inclined to say we should just remove it.

cc @vagababov @jonjohnsonjr

Reply:
Things look better with this removed...

Reply:
Looks like a bug even without the changes in this PR. I can trigger it on master by editing the PA of any scaled-to-zero revision and triggering a reconcile. Tagged you on #5106